### PR TITLE
Project download and upload buttons

### PR DESCRIPTION
**Pull request by Tom-Szendrey**
# Description
Adding some buttons to allow for projects to be downloaded and uploaded.

# Testing
Create a project
Download it with the new dropdown option
Upload it via the home page's upload modal

image
image

### CODE REVIEW

File: packages/client/hmi-client/src/components/home/tera-project-menu.vue [hunk 1]
```diff
@@ -58,11 +60,31 @@ const copyMenuItem = {
 	}
 };
 
+const downloadMenuItem = {
+	label: 'Download this project',
+	icon: 'pi pi-clone',
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/home/tera-project-menu.vue:65]:
```suggestion
	icon: 'pi pi-download',
```

File: packages/client/hmi-client/src/page/Home.vue [hunk 2]
```diff
@@ -91,6 +91,7 @@
 										<span class="p-button-label">{{ slotProps.option.value }}</span>
 									</template>
 								</SelectButton>
+								<Button icon="pi pi-plus" label="Upload project" @click="openUploadProjectModal" />
```
Comment (shawnyama) [inline packages/client/hmi-client/src/page/Home.vue:94]:
```suggestion
								<Button icon="pi pi-upload" label="Upload project" @click="openUploadProjectModal" />
```

File: packages/client/hmi-client/src/page/Home.vue [hunk 3]
```diff
@@ -148,6 +149,15 @@
...
 			</section>
+
+			<!-- Upload project modal -->
+			<section>
+				<tera-upload-project-modal
+					:visible="isUploadProjectModalVisible"
+					@project-added="useProjects().getAll()"
+					@close="isUploadProjectModalVisible = false"
+				/>
+			</section>
```
Comment (shawnyama) [inline packages/client/hmi-client/src/page/Home.vue:160]: No need to wrap in a `<section>`

**Review by dvince2 — APPROVED** [review]
Back end looks good

File: packages/client/hmi-client/src/components/home/tera-project-menu.vue [hunk 4]
```diff
@@ -58,11 +60,31 @@ const copyMenuItem = {
...
+				a.remove();
+			}
+			isCopying.value = false;
+		}
+	}
+};
+
 const separatorMenuItem = { separator: true };
 const projectMenuItems = computed(() => {
 	const items = [] as any[];
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/home/tera-project-menu.vue:84]: If this is easy to add in try:

```suggestion
	const items: MenuItem[] = [];
```

and

```
import { MenuItem } from 'primevue/menuitem';
```

File: packages/client/hmi-client/src/components/home/tera-project-menu.vue [hunk 5]
```diff
@@ -58,11 +60,31 @@ const copyMenuItem = {
 	}
 };
 
+const downloadMenuItem = {
+	label: 'Download this project',
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/home/tera-project-menu.vue:64]:
```suggestion
	label: 'Download',
```

File: packages/client/hmi-client/src/components/project/tera-upload-project-modal.vue [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+				<section>
+					<label class="subheader">Add project here.</label>
+					<tera-drag-and-drop-importer
+						:accept-types="[AcceptedTypes.PROJECTCONFIG]"
+						:accept-extensions="[AcceptedExtensions.PROJECTCONFIG]"
+						:import-action="verifyProject"
+						:progress="progress"
+						@import-completed="importCompleted"
+						@imported-files-updated="(value) => (importedFiles = value)"
+					></tera-drag-and-drop-importer>
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/project/tera-upload-project-modal.vue:17]:
```suggestion
					/>
```

**Review by YohannParis — APPROVED** [review]

File: packages/client/hmi-client/src/page/Home.vue (hunk 3 above)
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/page/Home.vue]: Thank you!

**Pull request merged**